main.py

```diff
@@ -0,0 +1,49 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+import crud
+import models
+import schemas
+from database import SessionLocal, engine
+
+models.Base.metadata.create_all(bind=engine)
+
+app = FastAPI()
+
+
+def get_db():
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.post("/authors/", response_model=schemas.AuthorRead)
+def create_author(author: schemas.AuthorCreate, db: Session = Depends(get_db)):
+    return crud.create_author(db=db, author=author)
+
+
+@app.get("/authors/{author_id}", response_model=schemas.AuthorRead)
+def read_author(author_id: int, db: Session = Depends(get_db)):
+    db_author = crud.get_author(db, author_id=author_id)
+    if db_author is None:
+        raise HTTPException(status_code=404, detail="Author not found")
+    return db_author
+
+
+@app.post("/books/", response_model=schemas.BookRead)
+def create_book(book: schemas.BookCreate, db: Session = Depends(get_db)):
+    return crud.create_book(db=db, book=book)
+
+
+@app.get("/books/", response_model=list[schemas.BookRead])
+def read_books(skip: int = 0, limit: int = 100, db: Session = Depends(get_db)):
+    books = crud.get_books(db, skip=skip, limit=limit)
+    return books
+
+
+@app.get("/books/by_author/{author_id}", response_model=list[schemas.BookRead])
```

Comment:
all paths shoud be with / in the end
```suggestion
@app.get("/books/by_author/{author_id}/", response_model=list[schemas.BookRead])
```